consumer/nf_discovery.go

```diff
@@ -10,29 +10,62 @@ import (
 	"fmt"
 	"net/http"
 
+	ausf_context "github.com/omec-project/ausf/context"
 	"github.com/omec-project/ausf/logger"
+	nrf_cache "github.com/omec-project/nrf/nrfcache"
 	"github.com/omec-project/openapi/Nnrf_NFDiscovery"
 	"github.com/omec-project/openapi/models"
 )
 
 func SendSearchNFInstances(nrfUri string, targetNfType, requestNfType models.NfType,
-	param Nnrf_NFDiscovery.SearchNFInstancesParamOpts) (*models.SearchResult, error) {
+	param *Nnrf_NFDiscovery.SearchNFInstancesParamOpts,
+) (models.SearchResult, error) {
+	if ausf_context.GetSelf().EnableNrfCaching {
+		return nrf_cache.SearchNFInstances(nrfUri, targetNfType, requestNfType, param)
+	} else {
+		return SendNfDiscoveryToNrf(nrfUri, targetNfType, requestNfType, param)
+	}
+}
+
+func SendNfDiscoveryToNrf(nrfUri string, targetNfType, requestNfType models.NfType,
+	param *Nnrf_NFDiscovery.SearchNFInstancesParamOpts,
+) (models.SearchResult, error) {
+	// Set client and set url
 	configuration := Nnrf_NFDiscovery.NewConfiguration()
 	configuration.SetBasePath(nrfUri)
 	client := Nnrf_NFDiscovery.NewAPIClient(configuration)
 
-	result, rsp, rspErr := client.NFInstancesStoreApi.SearchNFInstances(context.TODO(),
-		targetNfType, requestNfType, &param)
-	if rspErr != nil {
-		return nil, fmt.Errorf("NFInstancesStoreApi Response error: %+w", rspErr)
+	result, res, err := client.NFInstancesStoreApi.SearchNFInstances(context.TODO(), targetNfType, requestNfType, param)
+	if res != nil && res.StatusCode == http.StatusTemporaryRedirect {
+		err = fmt.Errorf("Temporary Redirect For Non NRF Consumer")
```

Comment:
```suggestion
		err = fmt.Errorf("temporary redirect for non NRF consumer")
```
Error logs should not be capitalized.
https://go.dev/wiki/CodeReviewComments#error-strings